sdk/appconfiguration/Azure.Data.AppConfiguration/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,165 @@
+using AzureEventSourceListener listener = AzureEventSourceListener.CreateConsoleLogger();
+```
+
+#### Enable content logging
+
+By default only URI and headers are logged. To enable content logging, set the `Diagnostics.IsLoggingContentEnabled` client option. 
+
+``` c#
+ConfigurationClientOptions options = new ConfigurationClientOptions()
+{
+    Diagnostics =
+    {
+        IsLoggingContentEnabled = true
+    }
+};
+```
+#### Logging redacted headers and query parameters
+
+Some sensitive headers and query parameters are not logged by default and are displayed as "REDACTED", to include them in logs use the `Diagnostics.LoggedHeaderNames` and `Diagnostics.LoggedQueryParameters` client options.
+
+``` c#
+ConfigurationClientOptions options = new ConfigurationClientOptions()
+{
+    Diagnostics =
+    {
+        LoggedHeaderNames = { "x-ms-request-id" },
+        LoggedQueryParameters = { "api-version" }
+    }
+};
+```
+
+You can also disable redaction completely by adding a "*" to collections mentioned above.
+
+``` c#
+ConfigurationClientOptions options = new ConfigurationClientOptions()
+{
+    Diagnostics =
+    {
+        LoggedHeaderNames = { "*" },
+        LoggedQueryParameters = { "*" }
+    }
+};
+```
+
+To learn about other logging mechanisms, see [Azure SDK diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md).
+
+## Troubleshooting authentication issues
+
+Azure App Configuration supports Azure Active Directory authentication. To provide a valid credential, you can use the `Azure.Identity` package. For more information on getting started, see the [Azure App Configuration library's README](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/appconfiguration/Azure.Data.AppConfiguration#authenticate-the-client). For details on the credential types supported in `Azure.Identity`, see the [Azure Identity library's documentation](https://docs.microsoft.com/dotnet/api/overview/azure/Identity-readme).
+
+Here are the authentication exceptions and ways to handle it:
+
+### AuthenticationFailedException
+
+Exceptions arising from authentication errors can be raised on any service client method that makes a request to the service. This is because the token is requested from the credential on the first call to the service and on any subsequent requests to the service that need to refresh the token. 
+
+To distinguish these failures from failures in the service client, Azure Identity classes raise the `AuthenticationFailedException` with details describing the source of the error in the exception message and possibly the error message. Depending on the application, these errors may or may not be recoverable.
+
+``` c#
+using Azure.Identity;
+using Azure.Data.AppConfiguration;
+
+// Create a ConfigurationClient using the DefaultAzureCredential
+string endpoint = "<endpoint>";
+ConfigurationClient client = new ConfigurationClient(new Uri(endpoint), new DefaultAzureCredential());
```

Comment:
```suggestion
string endpoint = "<endpoint>";

var client = new ConfigurationClient(new Uri(endpoint), new DefaultAzureCredential());
```
We should be consistent with how we're creating the client across snippets.   Here, we've got the type name.  In the other blocks, we're using `var`.